# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1371050393 14400
#      Wed Jun 12 11:19:53 2013 -0400
# Node ID d56cbc17566e58a22bc098bbbdcabc893f632b43
# Parent  787168f5f858b52492ee8e3e987482db015c1ada
avoid looking at previous token that doesn't exist (bug #39180)

* lex.ll (lexical_feedback::maybe_mark_previous_token_as_variable):
Check that tok is not null.

diff --git a/libinterp/parse-tree/lex.ll b/libinterp/parse-tree/lex.ll
--- a/libinterp/parse-tree/lex.ll
+++ b/libinterp/parse-tree/lex.ll
@@ -1624,17 +1624,18 @@ lexical_feedback::previous_token_may_be_
   const token *tok = tokens.front ();
   return tok ? tok->may_be_command () : false;
 }
 
 void
 lexical_feedback::maybe_mark_previous_token_as_variable (void)
 {
   token *tok = tokens.front ();
-  if (tok->is_symbol ())
+
+  if (tok && tok->is_symbol ())
     pending_local_variables.insert (tok->symbol_name ());
 }
 
 void
 lexical_feedback::mark_as_variables (const std::list<std::string>& lst)
 {
   for (std::list<std::string>::const_iterator p = lst.begin ();
        p != lst.end (); p++)
